gstreamer/src/buffer.rs

```diff
@@ -579,6 +579,79 @@ lazy_static! {
         *BUFFER_COPY_METADATA | ::BufferCopyFlags::MEMORY;
 }
 
+#[cfg(feature = "ser_de")]
+pub(crate) mod serde {
+    use serde::de::{Deserialize, Deserializer};
+    use serde::ser::{Serialize, Serializer, SerializeStruct};
+    use serde_bytes::{Bytes, ByteBuf};
+
+    use Buffer;
+    use BufferFlags;
+    use BufferRef;
+    use ClockTime;
+
+    impl<'a> Serialize for BufferRef {
+        fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+            let mut datetime = serializer.serialize_struct("Buffer", 6)?;
+            datetime.serialize_field("pts", &self.get_pts())?;
+            datetime.serialize_field("dts", &self.get_dts())?;
+            datetime.serialize_field("duration", &self.get_duration())?;
+            datetime.serialize_field("offset", &self.get_offset())?;
+            datetime.serialize_field("offset_end", &self.get_offset_end())?;
+            datetime.serialize_field("flags", &self.get_flags())?;
+            {
+                let data = self.map_readable().unwrap();
```

Comment:
Instead of panicking, this should fail cleanly I guess

Reply:
Sure thing!